packages/core/src/lib/modal.svelte

```diff
@@ -26,81 +26,90 @@ Creates a modal overlay.
 
 <script lang="ts">
 import cx from 'classnames';
-import { createEventDispatcher, onMount } from 'svelte';
 import IconButton from './button/icon-button.svelte';
+import { clickOutside } from '$lib';
+import type { Writable } from 'svelte/store';
+import { onMount } from 'svelte';
 
 /** Whether the modal is open. */
-export let open = false;
+export let isOpen: Writable<boolean>;
+
+$: if (typeof document !== 'undefined') {
+  document.body.classList.toggle('overflow-hidden', $isOpen);
+}
+
 /** The variant of the modal. */
 export let variant: 'small' | '' = '';
 
-const dispatch = createEventDispatcher<{
-  /** Fires when the modal is requested to close. */
-  close: boolean;
-}>();
-
 const handleCloseModal = () => {
-  dispatch('close', true);
-};
-
-const handleBackgroundClick = (event: MouseEvent) => {
-  if (event.target === event.currentTarget) {
-    handleCloseModal();
+  if ($isOpen) {
+    isOpen.set(false);
   }
 };
 
-const handleBackgroundKey = (event: KeyboardEvent) => {
-  if (event.key === 'Enter' || event.key === 'Space') {
+const handleEscapePress = (event: KeyboardEvent) => {
+  if (event.key === 'Escape') {
+    event.preventDefault();
     handleCloseModal();
   }
 };
 
 onMount(() => {
-  document.body.classList.toggle('overflow-hidden', open);
+  if ($isOpen) {
```

Comment:
- Attaching this login to `onMount` won't get us the behavior we want; we need it to run whenever the modal opens, which will likely occur more than once sometime after mount
- There's very rarely a need to use `querySelector` in Svelte, instead we can [`bind`]()  the `<h2>`'s DOM element to a `let` variable: https://svelte.dev/docs/element-directives#bind-this

We can use a reactive statement to achieve this behavior:

```ts
let headingElement: HTMLElement | undefined

$: headingElement?.focus()
```

The `$:` will cause the code that follows - `headingElement?.focus()` - to run whenever the `let headingElement` variable changes. The variable will change value in reaction to two events:

1. The modal appears and the `<h2>` element we're bound to mounts, so `headingElement` is set to the `HTMLElement`
2. The modal disappears and the `<h2>` unmounts, so `headingElement` is set to `undefined`

In (1), our statement will trigger focus. In (2), the statement will short-circuit.

Be sure to test this out in a browser! There's a chance this will call `.focus()` too early, and the element won't actually focus. If that's the case, then we may need to tweak this ever so slightly:

```ts
$: if (headingElement) {
  requestAnimationFrame(() => headingElement?.focus())
}
```

Reply:
Ahhh, I see - thanks so much for the explanation!